### PR TITLE
Add option to disable resource cache

### PR DESCRIPTION
**Pull request by karina-thompson**
We want an option to be able to turn off the resource cache for use in tests

### CODE REVIEW

File: lib/zendesk_api/configuration.rb [hunk 1]
```diff
@@ -42,8 +42,12 @@ class Configuration
     # @return [ZendeskAPI::LRUCache]
     attr_accessor :cache
 
+    # @ereturn [Boolean] Whether to use resource_cache or not
```
Comment (dferdian) [inline lib/zendesk_api/configuration.rb:45]: Small typo 

```suggestion
    # @return [Boolean] Whether to use resource_cache or not
```

**Review by dferdian — APPROVED** [review]
Thanks!

File: lib/zendesk_api/client.rb [hunk 2]
```diff
@@ -40,6 +40,8 @@ def method_missing(method, *args, &block)
       method = method.to_s
       options = args.last.is_a?(Hash) ? args.pop : {}
 
+      return ZendeskAPI::Collection.new(self, method_as_class(method), options) unless config.use_resource_cache
```
Comment (KJTsanaktsidis) [inline lib/zendesk_api/client.rb:43]: I think this is missing the nil check on `method_as_class` from line 50:

```
raise "Resource for #{method} does not exist" unless @resource_cache[method][:class]
```

unless `ZendeskAPI::Collection#initialize` is also doing a nill-check on the `method_as_class` argument?

**Review by KJTsanaktsidis — COMMENTED** [review]
Other than that, LGTM, nice.

File: lib/zendesk_api/client.rb (hunk 2 above)
Comment (karina-thompson) [reply lib/zendesk_api/client.rb:43]: Good catch! Will fix

**Review by KJTsanaktsidis — APPROVED** [review]

**Review by andrewbigger — APPROVED** [review]

**Comment by nogates** [comment]
Thank you all :)

**Pull request merged**